### PR TITLE
fix(inbox): keep compose FAB from covering the last conversation tile

### LINKED ISSUE #5314: fix(inbox): keep compose FAB from covering the last conversation tile
Auto-created from PR #5313.

**PR:** https://github.com/divinevideo/divine-mobile/pull/5313
**Branch:** `fix/dm-floating-action-button-overlay`

### PR DESCRIPTION
**Pull request by hm21**
Closes #5314

## Summary

The compose FAB (`InboxFab`) overlays the conversation list inside a `Stack` at `bottom: 16`. The `ListView.builder` had no bottom padding, so when scrolled to the end the last `ConversationTile` sat underneath the 56pt-tall FAB and was partially hidden.

This reserves bottom padding on the conversation list equal to the FAB's footprint (FAB height 56 + 16pt inset top and bottom = 88), so the last tile always clears the FAB. The 16pt FAB inset is now shared via a `_kFabInset` constant so the position and the reserved padding can't drift apart.

| Before | After |
| ------ | ------ |
|   image    |       image  |

## Changes

- `inbox_view.dart`: add `_kFabInset` and `_kConversationListBottomInset` constants; apply `padding: EdgeInsets.only(bottom: _kConversationListBottomInset)` to the conversation `ListView.builder`; use `_kFabInset` for the FAB's `PositionedDirectional`.
- `inbox_view_test.dart`: add a widget test asserting the conversation list reserves enough bottom padding to clear the FAB.

## Test plan

- [x] `flutter analyze lib/screens/inbox/inbox_view.dart` — no issues
- [x] `flutter test test/screens/inbox/inbox_view_test.dart` — 19/19 pass

### CODE REVIEW

File: mobile/lib/screens/inbox/inbox_view.dart [hunk 1]
```diff
@@ -191,6 +191,14 @@ class _InboxViewState extends ConsumerState<InboxView>
 /// pane width — the subtle shared-axis slide.
 const double _kTabSlideFraction = 0.06;
 
+/// Inset of the compose FAB from the bottom-right of the Messages pane.
+const double _kFabInset = 16;
+
+/// Bottom padding reserved on the conversation list so the FAB (56pt tall,
+/// inset [_kFabInset] from the bottom) never covers the last conversation
+/// tile when scrolled to the end. Equals the FAB's footprint plus a margin.
+const double _kConversationListBottomInset = _kFabInset + 56 + _kFabInset;
```
Comment (realmeylisdev) [inline mobile/lib/screens/inbox/inbox_view.dart]: **[should-fix]** The `56` here is a bare literal that duplicates the FAB's actual height. `InboxFab` hardcodes `SizedBox(width: 56, height: 56)` (`inbox_fab.dart:21-22`) and exports no public size constant (only a private `_borderRadius`), so the two `56`s are linked only by the doc comment just above ("the FAB (56pt tall …)"). The 88pt clearance is therefore correct only while the FAB stays exactly 56pt; if anyone resizes the FAB, this reserved padding silently goes stale and the exact bug this PR fixes reappears — with no compile error and no failing test (the test only asserts `>= 72`).

Suggest a single source of truth: add `static const double size = 56;` to `InboxFab`, use it inside its own `SizedBox`, and write `_kConversationListBottomInset = _kFabInset + InboxFab.size + _kFabInset;` (referencing `InboxFab.size` in the test too). Maps to `code_style.md` "No Hardcoded Values". Not a blocker — the geometry is correct today — but it locks the invariant in code rather than in a comment.

File: mobile/test/screens/inbox/inbox_view_test.dart [hunk 2]
```diff
@@ -298,6 +298,47 @@ void main() {
...
+          final listView = tester.widget<ListView>(
+            find.ancestor(
+              of: find.byType(ConversationTile),
+              matching: find.byType(ListView),
+            ),
+          );
+          final padding = listView.padding! as EdgeInsets;
+          // FAB is 56pt tall and inset 16pt from the bottom, so the list must
+          // reserve at least that much to keep the last tile fully visible.
+          expect(padding.bottom, greaterThanOrEqualTo(56 + 16));
```
Comment (realmeylisdev) [inline mobile/test/screens/inbox/inbox_view_test.dart]: **[should-fix]** This test is titled "… the FAB does not cover the last tile" but only asserts a static property: it reads `listView.padding` and checks `padding.bottom >= 56 + 16`. It never scrolls to the end, never reads the last tile's rect, and never compares it to the FAB's rect. Because `56 + 16` is a third uncoupled copy of the FAB height (after `inbox_fab.dart` and the production constant), the assertion is decoupled from the FAB itself: grow the FAB to 80 → footprint `16 + 80 = 96 > 88`, the last tile is covered again, yet `padding.bottom` stays `88` and `88 >= 72` passes green. So the test cannot catch the one regression vector it exists to guard.

Per `testing.md` "test behavior, not properties," prefer a geometry assertion — pump enough conversations to overflow, scroll to the bottom (the sibling "dispatches load more" test already drags this list `-5000`), and assert the last `ConversationTile`'s bottom edge sits at/above the FAB's top edge via `tester.getRect`. It does usefully guard against the padding regressing to zero, so treat this as a strengthening suggestion, not a defect in current behavior.

**Review by realmeylisdev — COMMENTED** [review]
## Formal review — PR #5313

Small, correct presentation-layer geometry fix. I verified it end-to-end against the source on the PR branch (`d1feb59`):

- **The math clears the FAB.** The conversation list's viewport bottom coincides with the `_MessagesContent` Stack bottom (the list lives in the `Expanded` of the only sibling `Column`). The FAB at `PositionedDirectional(bottom: 16)` with height `56` has its top edge `72pt` above that bottom. The reserved `_kConversationListBottomInset = 16 + 56 + 16 = 88` means the last tile clears the FAB top by a `16pt` margin when scrolled to the end. This holds for all three shapes the padded `ListView` renders (conversations only, banner + conversations, and the trailing load-more spinner).
- **Tests and analyze pass.** 19/19 in `inbox_view_test.dart` green here; `flutter analyze` clean on both touched files.
- **Pagination is unaffected.** `ScrollPaginationMixin` triggers `onLoadMore` at `maxScrollExtent - 200`; `88 < 200`, so the near-bottom trigger still fires (the existing load-more test stays green).
- **No backend surface.** Pure Flutter layout values — no client / repository / Nostr / relay / API code touched, so no backend repo in the org is affected by this change.
- **Tight blast radius.** The new constants are file-private (`_k` prefix), referenced only in `inbox_view.dart`; the empty-state and banner-only paths correctly receive no padding (their content is centered via `InboxEmptyState`'s `Center` and never reaches the FAB corner); there are no inbox golden files to go stale; `SafeArea(bottom: false)` is pre-existing and untouched.

Nice touch extracting the FAB inset into the named, documented `_kFabInset` and deriving the reserved padding from it.

## Findings

### Should-fix (non-blocking)

- **The FAB's `56pt` height is hardcoded in three unconnected places** — `InboxFab` (`SizedBox(width: 56, height: 56)`, no exported constant), the `_kConversationListBottomInset` production constant, and the test's `56 + 16` threshold. The coupling exists only in a doc comment, so resizing the FAB would silently stale the reserved padding and reintroduce this exact bug with no compile error and no failing test. See the inline comment on `inbox_view.dart`. (`code_style.md` → "No Hardcoded Values".)
- **The padding test asserts a property, not the behavior its title claims.** It reads `listView.padding` and checks `>= 72` rather than scrolling to the end and verifying the last `ConversationTile` clears the FAB's top edge. See the inline comment on `inbox_view_test.dart` for the regression vector it can't catch. (`testing.md` → "test behavior, not properties".)

### Observations (no action required in this PR)

- Empty-state and banner-only paths intentionally get no padding; their content is centered and never lands under the FAB. Pre-existing and correct.
- `InboxFab`'s `GestureDetector` (`inbox_fab.dart:23`) has no `Semantics`/tooltip — an accessibility gap (`accessibility.md` treats this as Must-Fix for interactive elements), so a screen reader gets no label and may not announce it as a button. That file is **not touched by this diff**, so it pre-exists on `main` and is out of scope here; flagging only as a follow-up (wrap the `GestureDetector` in `Semantics(button: true, label: <compose label>)` or use `DivineIconButton`). It's also the natural home for the `static const double size` referenced below.
- The raw `16` inset is **not** a spacing-token violation — this repo has no `AppSpacing` scale and raw `EdgeInsets` literals are the house style; the named `_kFabInset` is more disciplined than the surrounding code.

Verdict: **Comment** — nothing in the diff is a correctness or regression blocker; the two should-fix items are maintainability / test-robustness improvements.

File: mobile/lib/screens/inbox/inbox_view.dart (hunk 1 above)
Comment (hm21) [reply mobile/lib/screens/inbox/inbox_view.dart]: Done in 4eaec9590. Added `static const double size = 56` to `InboxFab` as the single source of truth: the FAB's own `SizedBox` now uses it, and `_kConversationListBottomInset = _kFabInset + InboxFab.size + _kFabInset` derives from it, so resizing the FAB tracks the reserved padding automatically — the invariant is now in code, not a comment.

File: mobile/test/screens/inbox/inbox_view_test.dart (hunk 2 above)
Comment (hm21) [reply mobile/test/screens/inbox/inbox_view_test.dart]: Done in 4eaec9590. Replaced the property assertion with a behavior/geometry one: the test now pumps 30 conversations, jumps the list to `maxScrollExtent`, and asserts the bottommost `ConversationTile`'s bottom edge sits at/above the `InboxFab`'s top edge (`tester.getRect`). With zero padding the last tile would reach the stack bottom while the FAB top is 72pt higher, so this now fails on exactly the regression vector the old `>= 72` check missed.

**Comment by hm21** [comment]
@realmeylisdev thanks for the thorough review — both should-fix items addressed in 4eaec9590:

1. **Single source of truth for the FAB size** — `InboxFab.size` is now a public constant used by the FAB's own `SizedBox` and by `_kConversationListBottomInset = _kFabInset + InboxFab.size + _kFabInset`. Resizing the FAB now propagates to the reserved padding automatically; the invariant lives in code, not a doc comment.
2. **Test now asserts behavior, not a property** — it pumps 30 conversations, jumps to `maxScrollExtent`, and asserts the last `ConversationTile`'s bottom edge clears the `InboxFab`'s top edge via `tester.getRect`. This fails on the FAB-grows / padding-shrinks regression the old `padding.bottom >= 72` check couldn't catch.

The out-of-scope `InboxFab` accessibility gap (no `Semantics`/tooltip on the `GestureDetector`) is left for a follow-up as you suggested, since that file isn't touched by this diff. `flutter analyze` clean and all 21 inbox tests green.

File: mobile/lib/screens/inbox/widgets/inbox_fab.dart [hunk 3]
```diff
@@ -13,13 +13,18 @@ class InboxFab extends StatelessWidget {
 
   final VoidCallback onPressed;
 
+  /// Diameter of the circular FAB. Single source of truth — layouts that
+  /// reserve space for the FAB (e.g. the conversation list bottom inset)
+  /// derive from this rather than re-hardcoding the value.
+  static const double size = 56;
```
Comment (realmeylisdev) [inline mobile/lib/screens/inbox/widgets/inbox_fab.dart:19]: Nit (non-blocking): this nails the single source of truth, but the class doc comment at line 9 still says “56x56 circle” — now the last hardcoded copy of the size. If `size` ever changes, that comment silently goes stale. Worth rewording to reference `[size]` (or dropping the literal dimensions).

**Review by realmeylisdev — APPROVED** [review]
Both should-fix items addressed — verified end-to-end on `4eaec9590`.

- **FAB size single source of truth** — `InboxFab.size` now drives both the FAB's own `SizedBox` and `_kConversationListBottomInset`. The clearance is `+_kFabInset` for any size (the `size` term cancels), so the padding can't drift from the FAB again.
- **Test asserts behavior, not a property** — confirmed it goes red on the actual regression: removing the inset and growing the FAB past the reserved space both fail the new geometry assertion, which the old `>= 72` check couldn't catch.

`flutter analyze` clean, inbox tests green. Approving. One trivial non-blocking nit inline. Agreed on deferring the `InboxFab` Semantics/tooltip gap to a follow-up.

File: mobile/lib/screens/inbox/widgets/inbox_fab.dart (hunk 3 above)
Comment (hm21) [reply mobile/lib/screens/inbox/widgets/inbox_fab.dart:19]: Good catch — fixed in 8f4771eef. The class doc now reads "A `[size]×[size]` circle", so it references the constant instead of carrying the last hardcoded literal.
